[Bug]: UI Crash/Blank screen - "input.trim is not a function" in ChatHistoryComponent (IndexedDB corruption)

If the browser's IndexedDB gets corrupted or wiped incorrectly, the RemixAI plugin fails to load the chat history properly. Instead of handling the null/undefined value gracefully, it attempts to call .trim() on it, causing a fatal crash that leaves the Remix IDE completely blank/frozen.

Steps to Reproduce:

Have a corrupted indexedDbCache (or simulate it).

Open Remix IDE with the RemixAI Plugin enabled.

The UI attempts to load ChatHistoryComponent.

The entire IDE crashes.

Expected Behavior:
The component should use defensive programming (e.g., if (typeof input !== 'string') return "";) to handle empty or corrupted histories gracefully and render an empty chat, instead of crashing the whole application.

Console Logs : 
SES Removing unpermitted intrinsics

main.c961d6aa047da0f1.js:1 [Matomo][AutoInit] Config setup completed

main.c961d6aa047da0f1.js:1 [Matomo][AutoInit] Auto-initializing with existing settings, perf analytics: true

main.c961d6aa047da0f1.js:1 [Matomo][AutoInit] Auto-initialized with immediate (cookie) mode

main.c961d6aa047da0f1.js:1 [Matomo][AutoInit] Pre-init queue processed

main.c961d6aa047da0f1.js:1 indexedDB is loaded...



la primera vez que entre si me funciono ...

main.c961d6aa047da0f1.js:1 localstorage is loaded...

main.c961d6aa047da0f1.js:1 indexedDB activated

1713.b04e9ed61b3ca1c0.js:1 IndexedDB Cache: Database ready

main.c961d6aa047da0f1.js:1 Uncaught Error: Plugin "indexedDbCache" is currently deactivated. It cannot emit the event cacheReady. Activate "indexedDbCache" first.

    at plugin.emit (main.c961d6aa047da0f1.js:1:4612001)

    at request.onsuccess (1713.b04e9ed61b3ca1c0.js:1:2215196)

1713.b04e9ed61b3ca1c0.js:1 Error getting current file: Error: Error: No such file or directory No file selected

    at FileManager.file (1713.b04e9ed61b3ca1c0.js:1:3274115)

    at FileManager.callPluginMethod (main.c961d6aa047da0f1.js:1:6204801)

    at FileManager.<anonymous> (main.c961d6aa047da0f1.js:1:6205500)

    at Generator.next (<anonymous>)

    at main.c961d6aa047da0f1.js:1:2316407

    at new Promise (<anonymous>)

    at Module.__awaiter (main.c961d6aa047da0f1.js:1:2316045)

    at queue.callMethod (main.c961d6aa047da0f1.js:1:6205447)

    at PluginQueueItem.<anonymous> (main.c961d6aa047da0f1.js:1:3529125)

    at Generator.next (<anonymous>)

_callee10$ @ 1713.b04e9ed61b3ca1c0.js:1

1713.b04e9ed61b3ca1c0.js:1 Initializing Remix MCP server

1713.b04e9ed61b3ca1c0.js:5 [getWorkspaces] Failed to retrieve workspaces: TypeError: Cannot read properties of undefined (reading 'fileProviders')

    at workspaces (1713.b04e9ed61b3ca1c0.js:5:66242)

    at new Promise (<anonymous>)

    at _callee39$ (1713.b04e9ed61b3ca1c0.js:5:66179)

    at main.c961d6aa047da0f1.js:1:2465438

    at Generator.<anonymous> (main.c961d6aa047da0f1.js:1:7831574)

    at Generator.next (main.c961d6aa047da0f1.js:1:2175593)

    at asyncGeneratorStep (main.c961d6aa047da0f1.js:1:376896)

    at _next (main.c961d6aa047da0f1.js:1:377118)

    at main.c961d6aa047da0f1.js:1:377234

    at new Promise (<anonymous>)

_callee39$ @ 1713.b04e9ed61b3ca1c0.js:5

1713.b04e9ed61b3ca1c0.js:5 [getWorkspaces] Failed to retrieve workspaces: TypeError: Cannot read properties of undefined (reading 'fileProviders')

    at workspaces (1713.b04e9ed61b3ca1c0.js:5:66242)

    at new Promise (<anonymous>)

    at _callee39$ (1713.b04e9ed61b3ca1c0.js:5:66179)

    at main.c961d6aa047da0f1.js:1:2465438

    at Generator.<anonymous> (main.c961d6aa047da0f1.js:1:7831574)

    at Generator.next (main.c961d6aa047da0f1.js:1:2175593)

    at asyncGeneratorStep (main.c961d6aa047da0f1.js:1:376896)

    at _next (main.c961d6aa047da0f1.js:1:377118)

    at main.c961d6aa047da0f1.js:1:377234

    at new Promise (<anonymous>)

_callee39$ @ 1713.b04e9ed61b3ca1c0.js:5

1713.b04e9ed61b3ca1c0.js:1 Loading https://binaries.soliditylang.org/wasm/soljson-v0.8.34+commit.80d5c536.js with worker

1713.b04e9ed61b3ca1c0.js:1 Loading https://binaries.soliditylang.org/wasm/soljson-v0.8.34+commit.80d5c536.js?t=1776813935944 with worker

1713.b04e9ed61b3ca1c0.js:5 setting workspace Object

main.c961d6aa047da0f1.js:1 TypeError: input.trim is not a function

    at normalizeMarkdown (main.c961d6aa047da0f1.js:1:565181)

    at RemixMarkdownViewer (1713.b04e9ed61b3ca1c0.js:1:4575122)

    at 1713.b04e9ed61b3ca1c0.js:1:4566708

    at Array.map (<anonymous>)

    at ChatHistoryComponent (1713.b04e9ed61b3ca1c0.js:1:4565701)

    at Xh (main.c961d6aa047da0f1.js:1:962379)

    at Wk (main.c961d6aa047da0f1.js:1:1015235)

    at Vk (main.c961d6aa047da0f1.js:1:1010924)

    at Uk (main.c961d6aa047da0f1.js:1:1010855)

    at Jk (main.c961d6aa047da0f1.js:1:1010714)

Mi @ main.c961d6aa047da0f1.js:1

main.c961d6aa047da0f1.js:1 Uncaught TypeError: input.trim is not a function

    at normalizeMarkdown (main.c961d6aa047da0f1.js:1:565181)

    at RemixMarkdownViewer (1713.b04e9ed61b3ca1c0.js:1:4575122)

    at 1713.b04e9ed61b3ca1c0.js:1:4566708

    at Array.map (<anonymous>)

    at ChatHistoryComponent (1713.b04e9ed61b3ca1c0.js:1:4565701)

    at Xh (main.c961d6aa047da0f1.js:1:962379)

    at Wk (main.c961d6aa047da0f1.js:1:1015235)

    at Vk (main.c961d6aa047da0f1.js:1:1010924)

    at Uk (main.c961d6aa047da0f1.js:1:1010855)

    at Jk (main.c961d6aa047da0f1.js:1:1010714)

main.c961d6aa047da0f1.js:1 TypeError: Cannot set properties of null (setting 'currentContent')

    at ref (1713.b04e9ed61b3ca1c0.js:1:1747473)

    at Mj (main.c961d6aa047da0f1.js:1:993418)

    at ak (main.c961d6aa047da0f1.js:1:997065)

    at Zj (main.c961d6aa047da0f1.js:1:996905)

    at ak (main.c961d6aa047da0f1.js:1:997100)

    at Zj (main.c961d6aa047da0f1.js:1:996905)

    at ak (main.c961d6aa047da0f1.js:1:997665)

    at Zj (main.c961d6aa047da0f1.js:1:996905)

    at ak (main.c961d6aa047da0f1.js:1:997986)

    at Zj (main.c961d6aa047da0f1.js:1:996905)

Mi @ main.c961d6aa047da0f1.js:1

main.c961d6aa047da0f1.js:1 TypeError: The "listener" argument must be of type Function. Received type undefined

    at checkListener (main.c961d6aa047da0f1.js:1:5217205)

    at EventEmitter.removeListener (main.c961d6aa047da0f1.js:1:5221229)

    at main.c961d6aa047da0f1.js:1:7541906

    at Nj (main.c961d6aa047da0f1.js:1:993487)

    at Qj (main.c961d6aa047da0f1.js:1:995201)

    at Ik (main.c961d6aa047da0f1.js:1:1012834)

    at Fk (main.c961d6aa047da0f1.js:1:1007784)

    at jg (main.c961d6aa047da0f1.js:1:945984)

    at main.c961d6aa047da0f1.js:1:1005418

Mi @ main.c961d6aa047da0f1.js:1

main.c961d6aa047da0f1.js:1 Uncaught TypeError: Cannot set properties of null (setting 'currentContent')

    at ref (1713.b04e9ed61b3ca1c0.js:1:1747473)

    at Mj (main.c961d6aa047da0f1.js:1:993418)

    at ak (main.c961d6aa047da0f1.js:1:997065)

    at Zj (main.c961d6aa047da0f1.js:1:996905)

    at ak (main.c961d6aa047da0f1.js:1:997100)

    at Zj (main.c961d6aa047da0f1.js:1:996905)

    at ak (main.c961d6aa047da0f1.js:1:997665)

    at Zj (main.c961d6aa047da0f1.js:1:996905)

    at ak (main.c961d6aa047da0f1.js:1:997986)

    at Zj (main.c961d6aa047da0f1.js:1:996905)

1713.b04e9ed61b3ca1c0.js:1 [RemixAI Plugin] MCP server connected: Remix IDE Server

1713.b04e9ed61b3ca1c0.js:1 [RemixAI Plugin] MCP server connected: Web Search

1713.b04e9ed61b3ca1c0.js:1 [RemixAI Plugin] MCP server connected: ethskills

1713.b04e9ed61b3ca1c0.js:1 [RemixAI Plugin] MCP server connected: OpenZeppelin Contracts Cairo

1713.b04e9ed61b3ca1c0.js:1 [RemixAI Plugin] MCP server connected: OpenZeppelin Contracts

api.remix.live/billing/config:1  Failed to load resource: the server responded with a status of 401 ()

1713.b04e9ed61b3ca1c0.js:1 [AuthPlugin] Failed to fetch Paddle config: Authentication required

_callee6$ @ 1713.b04e9ed61b3ca1c0.js:1

1713.b04e9ed61b3ca1c0.js:1 [RemixAI Plugin] MCP server connected: OpenZeppelin Contracts Stellar

1713.b04e9ed61b3ca1c0.js:1 [RemixAI Plugin] MCP server connected: OpenZeppelin Contracts Stylus

1713.b04e9ed61b3ca1c0.js:1 Storage monitor: Summary Object

--- a/apps/remix-ide-e2e/src/tests/chatHistory.test.ts
+++ b/apps/remix-ide-e2e/src/tests/chatHistory.test.ts
@@ -419,7 +419,6 @@ module.exports = {
         const ids = result.value as string[]
         if (!ids || ids.length < 2) return
 
-        const newestId = ids[0]
         const oldestId = ids[1]
 
         browser
@@ -497,8 +496,8 @@ module.exports = {
         const lastBubble = bubbles[bubbles.length - 1]
         return lastBubble ? lastBubble.textContent : null
       }, [], function (result) {
-        browser.assert.ok(
-          result.value && (result.value as string).includes('<script>'),
+        //@ts-ignore
+        browser.assert.ok(result.value && (result.value as string).includes('<script>'),
           'Special characters are properly escaped'
         )
       })
@@ -700,5 +699,126 @@ module.exports = {
             'Prompt contains the current user request content'
           )
       })
-  },
+  },'Should show an emtpy bubble and not crash the UI when payload is null or undefined #group4': function (browser: NightwatchBrowser) {
+    const convId = 'null-content-crash-test-conv-001'
+
+    browser
+      // Seed a conversation whose assistant messages have null / missing content
+      // to simulate a corrupted or partially-written IndexedDB entry
+      .executeAsync(function (convId, done) {
+        const request = indexedDB.open('RemixAIChatHistory', 1)
+        request.onerror = () => done(false)
+        request.onsuccess = () => {
+          const db = request.result
+          const now = Date.now()
+          const tx = db.transaction(['conversations', 'messages'], 'readwrite')
+          tx.objectStore('conversations').put({
+            id: convId,
+            title: 'Null content crash regression',
+            preview: 'Valid user message',
+            createdAt: now,
+            updatedAt: now,
+            lastAccessedAt: now,
+            messageCount: 3,
+            archived: false
+          })
+          const msgStore = tx.objectStore('messages')
+          // Healthy user message — gives the conversation a real title/preview
+          msgStore.put({
+            id: convId + '-user-1',
+            role: 'user',
+            content: 'Valid user message',
+            timestamp: now,
+            conversationId: convId
+          })
+          // Corrupted assistant message: content explicitly null
+          msgStore.put({
+            id: convId + '-assistant-null',
+            role: 'assistant',
+            content: null,
+            timestamp: now + 1,
+            conversationId: convId
+          })
+          // Corrupted assistant message: content field entirely absent (undefined on read-back)
+          msgStore.put({
+            id: convId + '-assistant-missing',
+            role: 'assistant',
+            timestamp: now + 2,
+            conversationId: convId
+          })
+          tx.oncomplete = () => done(true)
+          tx.onerror = () => done(false)
+        }
+        request.onupgradeneeded = (event: any) => {
+          const db = event.target.result
+          if (!db.objectStoreNames.contains('conversations')) {
+            const store = db.createObjectStore('conversations', { keyPath: 'id' })
+            store.createIndex('createdAt', 'createdAt', { unique: false })
+            store.createIndex('archived', 'archived', { unique: false })
+            store.createIndex('lastAccessedAt', 'lastAccessedAt', { unique: false })
+          }
+          if (!db.objectStoreNames.contains('messages')) {
+            const msgStore = db.createObjectStore('messages', { keyPath: 'id' })
+            msgStore.createIndex('conversationId', 'conversationId', { unique: false })
+            msgStore.createIndex('timestamp', 'timestamp', { unique: false })
+          }
+        }
+      }, [convId], function (result) {
+        browser.assert.equal(result.value, true, 'Conversation with null/missing content seeded into IndexedDB')
+      })
+      .refreshPage()
+      .clickLaunchIcon('remixaiassistant')
+      .waitForElementPresent({
+        selector: "//*[@data-id='remix-ai-assistant-ready']",
+        locateStrategy: 'xpath',
+        timeout: 60000
+      })
+      // Open history sidebar and load the corrupted conversation
+      .waitForElementVisible('*[data-id="toggle-history-btn"]')
+      .click('*[data-id="toggle-history-btn"]')
+      .pause(1000)
+      .waitForElementVisible(`*[data-id="conversation-item-${convId}"]`, 5000)
+      .click(`*[data-id="conversation-item-${convId}"]`)
+      .pause(1000)
+      // Primary assertion: the IDE panel is still mounted — not blank or frozen.
+      // Before the fix this element was absent because the TypeError unmounted the tree.
+      .waitForElementPresent({
+        selector: "//*[@data-id='remix-ai-assistant-ready']",
+        locateStrategy: 'xpath',
+        timeout: 10000
+      })
+      // The chat bubble section must be present — proves React did not hard-crash
+      .waitForElementVisible('*[data-id="ai-response-chat-bubble-section"]', 5000)
+      // The valid user message must render with its text intact
+      .assert.textContains('*[data-id="ai-user-chat-bubble"]', 'Valid user message')
+      // Verify via DOM inspection that both panels survived and no uncaught error
+      // replaced the React root with an empty document body
+      .execute(function () {
+        const panelReady = document.querySelector('[data-id="remix-ai-assistant-ready"]')
+        const bubbleSection = document.querySelector('[data-id="ai-response-chat-bubble-section"]')
+        const allBubbles = document.querySelectorAll('.chat-bubble')
+        // Each assistant bubble with null/undefined content should render as an
+        // empty node — not throw, and not be completely absent
+        const assistantBubbles = Array.from(allBubbles).filter(el => el.classList.contains('bubble-assistant'))
+        return {
+          panelPresent: panelReady !== null,
+          bubbleSectionPresent: bubbleSection !== null,
+          assistantBubbleCount: assistantBubbles.length,
+          // Confirm the null-content bubbles did not inject "null" or "undefined" as literal text
+          noLiteralNull: assistantBubbles.every(b => b.textContent !== 'null' && b.textContent !== 'undefined')
+        }
+      }, [], function (result) {
+        const info = result.value as {
+          panelPresent: boolean
+          bubbleSectionPresent: boolean
+          assistantBubbleCount: number
+          noLiteralNull: boolean
+        }
+        browser
+          .assert.ok(info.panelPresent, 'AI assistant panel is present — UI did not crash on null content')
+          .assert.ok(info.bubbleSectionPresent, 'Chat bubble section rendered — no blank screen')
+          .assert.ok(info.assistantBubbleCount >= 1, `At least one assistant bubble rendered, got ${info.assistantBubbleCount}`)
+          .assert.ok(info.noLiteralNull, 'Null/undefined content rendered as empty string, not as literal "null"/"undefined"')
+      })
+  }
 }

--- a/libs/remix-ui/helper/src/lib/components/remix-md-renderer.tsx
+++ b/libs/remix-ui/helper/src/lib/components/remix-md-renderer.tsx
@@ -7,6 +7,9 @@ import remarkGfm from 'remark-gfm'
 import '../css/helper.css'
 
 export function normalizeMarkdown(input: string): string {
+  if (typeof input !== 'string') {
+    return ''
+  }
   let result = input
     .trim()
     .replace(/\n{2,}/g, '\n\n')

--- a/libs/remix-ui/remix-ai-assistant/src/components/chat.tsx
+++ b/libs/remix-ui/remix-ai-assistant/src/components/chat.tsx
@@ -112,7 +112,7 @@ export const ChatHistoryComponent: React.FC<ChatHistoryComponentProps> = ({
 
                     <div className="aiMarkup lh-base text-wrap">
                       {msg.role === 'assistant' ? (
-                        RemixMarkdownViewer(theme, msg.content)
+                        RemixMarkdownViewer(theme, msg.content ?? '')
                       ) : (
                         <div className="ai-paragraph pb-0">
                           {msg.content}

